File: Demo5/src/main/java/com/example/myapplication/Lambda.java
package com.example.myapplication;

/**
 * @author lzl
 * @ describe
 * @ time 2018/11/20 10:42
 */
public class Lambda {

    public static void main(String[] args){
        get((str, age)->{ System.out.print("");System.out.print("abc");});

    }

    public static void  get(TestInter testInter){

    }

    interface TestInter{
        void test(String str,int age);
    }
}

File: Demo6/src/main/java/com/example/myapplication/Lambda.kt
package com.example.myapplication

/**
 * @ describe
 *
 * @author lzl
 *
 * @ time 2018/11/20 18:49
 *
 *
 */

val printMsg = { msg: String ->
    println(msg)
}

fun main(args: Array<String>) {
    printMsg.invoke("hello")
}